tests/unit/test_charm.py

```diff
@@ -654,3 +654,51 @@ def test_postgres_relation_data(self):
                         self.harness.model.unit.status,
                         WaitingStatus("Waiting for database relation"),
                     )
+
+    def test_redis_relation_data(self):
+        """
+        arrange: given a deployed discourse charm and some relation data
+        act: add the redis relation to the charm
+        assert: the charm should wait for some correct relation data
+        """
+        test_cases = [
+            (
+                {"hostname": "redis-host", "port": 1010},
+                True,
+            ),
+            (
+                {"hostname": "", "port": 1010},
+                False,
+            ),
+            (
+                {"hostname": "redis-host", "port": None},
+                False,
+            ),
+            (
+                {"hostname": None, "port": None},
+                False,
+            ),
+            (
+                {},
+                False,
+            ),
+            (
+                {"port": 6379},
+                False,
+            ),
+            (
+                {"hostname": "redis-port"},
+                False,
+            ),
```

Comment:
I think you're missing test cases where the port is 0, but you're doing `int(redis_unit_data.get("port", 0) or 0)` so it could very well be 0. I think it'll work fine, but would be good to include in the test cases.

Reply:
I added a test case with port 0